src/pymortests/strategies.py

```diff
@@ -0,0 +1,448 @@
+        ni = draw(element_count)
+        return df.FunctionSpace(df.UnitSquareMesh(ni, ni), 'Lagrange', 1)
+
+    @hyst.composite
+    def fenics_vector_array(draw, count=1, dtype=None, length=None, compatible=True):
+        assume(compatible)
+        if dtype: # complex is not actually supported
+            assert dtype == np.float64
+        dtype = np.float64
+        lngs = draw(length or hyst.tuples(*[hy_lengths for _ in range(count)]))
+        V = draw(fenics_spaces())
+        Us = [FenicsVectorSpace(V).zeros(l) for l in lngs]
+        dims = [U.dim for U in Us]
+        for i in range(count):
+            # dtype is float here since the petsc vector is not setup for complex
+            for v, a in zip(Us[i]._list, draw(np_arrays(lngs[i], dims[i], dtype=dtype))):
+                v = FenicsVectorSpace(V).vector_from_numpy(a)
+        return Us
+else:
+    fenics_vector_array = nothing
+
+if config.HAVE_NGSOLVE:
+    @hyst.composite
+    def ngsolve_vector_array(draw, count=1, dtype=None, length=None, compatible=True):
+        if dtype: # complex is not actually supported
+            assert dtype == np.float64
+        dtype = np.float64
+        dim = draw(hy_dims(count, compatible))
+        lngs = draw(length or hyst.tuples(*[hy_lengths for _ in range(count)]))
+        spaces = [create_ngsolve_space(d) for d in dim]
+        Us = [s.zeros(l) for s,l in zip(spaces, lngs)]
+        for i in range(count):
+            for v, a in zip(Us[i]._list, draw(np_arrays(lngs[i], dim[i], dtype=dtype))):
+                v.to_numpy()[:] = a
+        return Us
+else:
+    ngsolve_vector_array = nothing
+
+if config.HAVE_DEALII:
+    @hyst.composite
+    def dealii_vector_array(draw, count=1, dtype=None, length=None, compatible=True):
+        dim = draw(hy_dims(count, compatible))
+        dtype = dtype or draw(hy_dtypes)
+        lngs = draw(length or hyst.tuples(*[hy_lengths for _ in range(count)]))
+        Us = [DealIIVectorSpace(d).zeros(l) for d, l in zip(dim, lngs)]
+        for i in range(count):
+            for v, a in zip(Us[i]._list, draw(np_arrays(lngs[i], dim[i], dtype=dtype))):
+                v.impl[:] = a
+        return Us
+else:
+    dealii_vector_array = nothing
+
+
+def picklable_vector_arrays(count=1, dtype=None, length=None, compatible=True):
+    return numpy_vector_array(count, dtype, length, compatible) | \
+           numpy_list_vector_array(count, dtype, length, compatible) | \
+           block_vector_array(count, dtype, length, compatible)
+
+
+# strategies shrink to the left while falsifying, make the "simplest" the leftmost
+def vector_arrays(count=1, dtype=None, length=None, compatible=True):
+    return picklable_vector_arrays(count, dtype, length, compatible) | \
+           fenics_vector_array(count, dtype, length, compatible) | \
+           dealii_vector_array(count, dtype, length, compatible) | \
+           ngsolve_vector_array(count, dtype, length, compatible)
```

Comment:
So if I understand this correctly, nothing will ensure that arrays from all available backends are tested for each individual test. I am not sure if I like that.